examples/vision/gradient_centralization.py

```diff
@@ -0,0 +1,253 @@
+"""
+Title: Gradient Centralization for Better Training Performance
+Author: [Rishit Dagli](https://github.com/Rishit-dagli)
+Date created: FILLME
+Last modified: FILLME
+Description: Implement Gradient Centralization to improve training performance of DNNs.
+"""
+"""
+# Introduction
+
+This example implements [Gradient Centralization](https://arxiv.org/abs/2004.01461), a
+new optimization technique for Deep Neural Networks by Yong et al., and demonstrates it
+on Laurence Moroney's [Horses or Humans
+Dataset](https://www.tensorflow.org/datasets/catalog/horses_or_humans). Gradient
+Centralization can both speedup training process and improve the final generalization
+performance of DNNs. It operates directly on gradients by centralizing the gradient
+vectors to have zero mean. Gradient Centralization morever improves the Lipschitzness of
+the loss function and its gradient so that the training process becomes more efficient
+and stable.
+
+This example requires TensorFlow 2.2 or higher, as well as a package I built,
+[gradient-centralization-tf](https://github.com/Rishit-dagli/Gradient-Centralization-Tenso
+rFlow), which can be installed using the following command:
+
+```py
+pip install gradient-centralization-tf
+```
+"""
+
+"""
+# Setup
+"""
+
+"""shell
+!pip install gradient-centralization-tf
+"""
+
+import tensorflow as tf
+import tensorflow_datasets as tfds
+from tensorflow.keras import layers
+from tensorflow.keras.layers import Conv2D
+from tensorflow.keras.layers import MaxPooling2D
+from tensorflow.keras.layers import Dropout
+from tensorflow.keras.layers import Dense
+from tensorflow.keras.optimizers import RMSprop
+
+
+import gctf
+
+from time import time
+import os
+
+"""
+# Prepare the data
+
+For this example, we will be using the [Horses or Humans
+dataset](https://www.tensorflow.org/datasets/catalog/horses_or_humans).
+"""
+
+num_classes = 2
+input_shape = (300, 300, 3)
+dataset_name = "horses_or_humans"
+
+(train_ds, test_ds), metadata = tfds.load(
+    name=dataset_name,
+    split=[tfds.Split.TRAIN, tfds.Split.TEST],
+    with_info=True,
+    as_supervised=True,
+)
+
+print(f"Image shape: {metadata.features['image'].shape}")
+print(f"Training images: {metadata.splits['train'].num_examples}")
+print(f"Test images: {metadata.splits['test'].num_examples}")
+
+"""
+# Use Data Augmentation
+
+We will rescale the data to `[0, 1]`  andperform simple augmentations to our data.
+"""
+
+rescale = tf.keras.Sequential([layers.experimental.preprocessing.Rescaling(1.0 / 255)])
```

Comment:
You don't need a Sequential model here. Just `rescale = layers.experimental.preprocessing.Rescaling(1.0 / 255)`

Reply:
Ah yes, thanks!